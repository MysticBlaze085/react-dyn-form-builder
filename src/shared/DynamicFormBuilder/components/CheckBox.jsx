import * as React from 'react';
import FormGroup from '@material-ui/core/FormGroup';
import Checkbox from '@material-ui/core/Checkbox';
import FormControlLabel from '@material-ui/core/FormControlLabel';

const CheckBox = ({ handler }) => {
  return(
  <div>
   <FormGroup row>
      <FormControlLabel
        control={<Checkbox />}
      />
      </FormGroup>
  </div>
)};

export default CheckBox;