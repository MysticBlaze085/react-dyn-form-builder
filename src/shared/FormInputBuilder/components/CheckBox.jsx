import React from 'react';
import FormLabel from '@material-ui/core/FormLabel';
import FormGroup from '@material-ui/core/FormGroup';
import FormControlLabel from '@material-ui/core/FormControlLabel';
import Checkbox from '@material-ui/core/Checkbox';

const Checkboxs = ({ handler, meta: { label, items } }) => {
    const availableItems = items.map((Item) => (
        <FormControlLabel
            key={Item.key}
            control={<Checkbox value={Item.key} label={Item.value} {...handler('checkbox')} />}
            label={Item.value}
        />
    ));
    return (
        <div style={{ width: '100%' }}>
            <FormLabel component="legend">{label}</FormLabel>
            <FormGroup>{availableItems}</FormGroup>
        </div>
    );
};

export default Checkboxs;
