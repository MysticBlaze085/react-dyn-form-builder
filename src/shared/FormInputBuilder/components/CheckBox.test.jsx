import React from 'react';
import { shallow } from 'enzyme';
import CheckBox from './CheckBox';

const MOCK_DATA = {
    formState: 'other',
    meta: {
        label: 'Gender',
        items: [
            { key: 'male', value: 'Male' },
            { key: 'female', value: 'Female' },
            { key: 'other', value: 'other' },
        ],
        classes: 'col-6',
    },
    options: {},
    handler: () => {},
};

it('expect CheckBox to render', () => {
    expect(shallow(<CheckBox {...MOCK_DATA} />)).toMatchSnapshot();
});
