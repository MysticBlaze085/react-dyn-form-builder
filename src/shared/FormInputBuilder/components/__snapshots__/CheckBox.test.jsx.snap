// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`expect CheckBox to render 1`] = `ShallowWrapper {}`;
